petastorm/fs_utils.py

```diff
@@ -139,3 +139,9 @@ def filesystem(self):
         :return: The pyarrow filesystem object
         """
         return self._filesystem
+
+    def filesystem_factory(self):
+        def factory():
+            resolver = FilesystemResolver(self._dataset_url)
```

Comment:
The object might have been created with additional parameters, not only the url (`self, dataset_url, hadoop_configuration=None, connector=HdfsConnector, hdfs_driver='libhdfs3')`). We need to reliably reconstruct this object with proper parameters.

Would be nice to see a unit test confirming this information is not lost.

Reply:
Removed this - the factory methods now don't re-execute `FilesystemResolver`'s constructor.